src/js/initializers/initZqdConnection.ts

```diff
@@ -3,26 +3,32 @@ import {initSpace} from "../flows/initSpace"
 import Clusters from "../state/Clusters"
 import Current from "../state/Current"
 import getUrlSearchParams from "../lib/getUrlSearchParams"
-import refreshSpaceNames from "../flows/refreshSpaceNames"
+import refreshConnectionsSpaceNames from "../flows/refreshConnectionsSpaceNames"
+import {globalDispatch} from "../state/GlobalContext"
 
 export default async function(store: Store) {
   const {space, host, port, id} = getUrlSearchParams()
   global.windowId = id
 
-  const clusterHost = host || "localhost"
-  const clusterPort = port || "9867"
-  const clusterId = `${clusterHost}:${clusterPort}`
-  const cluster = {
-    id: clusterId,
-    host: clusterHost,
-    port: clusterPort,
-    username: "",
-    password: ""
+  const lastConn = Current.getConnection(store.getState())
+  if ((port && host) || !lastConn) {
+    const clusterHost = host || "localhost"
+    const clusterPort = port || "9867"
+    const clusterId = `${clusterHost}:${clusterPort}`
+    const cluster = {
+      id: clusterId,
+      host: clusterHost,
+      port: clusterPort,
+      username: "",
+      password: ""
+    }
+
+    store.dispatch(Clusters.add(cluster))
```

Comment:
It looks like there is a case when we'll never get a cluster variable defined?

Is this the logic?

1. If the host/port was provided in the url params, create a connection from them, add it, then set it to current.
2. Else if there was a current connection already, do nothing more in this file
3. Else create a default host and port, add it, then set it to current

Actually, do we always need to create the default one here?

Reply:
Yes, exactly. And I think we do need the default for the first time anyone loads the app.

Reply:
This conditional logic was getting hard to understand at first glance. What do you think about a version like this where we extract some of the connection setup into a new function?

```ts

const setupConnection = (host, port) => (dispatch, _, {globalDispatch}) => {
  const cluster = {
    host,
    port,
    id: [host, port].join(":"),
    username: "",
    password: ""
  }
  dispatch(Clusters.add(cluster))
  globalDispatch(Clusters.add(cluster))
  dispatch(Current.setConnectionId(cluster.id))
}

export default async function(store: Store) {
  const {space, host, port, id} = getUrlSearchParams()
  global.windowId = id

  const existingConnection = Current.getConnection(store.getState())

  if (host && port) {
    store.dispatch(setupConnection(host, port))
  } else if (!existingConnection) {
    store.dispatch(setupConnection("localhost", "9867"))
  }

  await store.dispatch(refreshConnectionsSpaceNames())

  const spaceId = space || Current.getSpaceId(store.getState())

  if (spaceId) store.dispatch(initSpace(spaceId))
}
```

Reply:
Sure thing, will use this instead!